src/rpc/context.rs

```diff
@@ -0,0 +1,309 @@
+        let [start_height, end_height]: [u32; 2] = rpc_params.parse()?;
+        rpc_context
+            .get_block_hashes(start_height, end_height)
+            .map_err(JsonrpseeError::to_call_error)
+            .await
+    })?;
+
+    module.register_async_method("getblockheader", |rpc_params, rpc_context| async move {
+        let height = rpc_params.parse::<[u32; 1]>()?[0];
+        rpc_context.get_block_header(height).map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    module.register_async_method("getblocktemplate", |_rpc_params, rpc_context| async move {
+        rpc_context.get_block_template().map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    module.register_async_method("getblocktransactions", |rpc_params, rpc_context| async move {
+        let height = rpc_params.parse::<[u32; 1]>()?[0];
+        rpc_context
+            .get_block_transactions(height)
+            .map_err(JsonrpseeError::to_call_error)
+            .await
+    })?;
+
+    module.register_async_method("getciphertext", |rpc_params, rpc_context| async move {
+        let commitment = rpc_params.parse::<[N::Commitment; 1]>()?[0];
+        rpc_context.get_ciphertext(commitment).map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    module.register_async_method("getledgerproof", |rpc_params, rpc_context| async move {
+        let commitment = rpc_params.parse::<[N::Commitment; 1]>()?[0];
+        rpc_context
+            .get_ledger_proof(commitment)
+            .map_err(JsonrpseeError::to_call_error)
+            .await
+    })?;
+
+    module.register_async_method("getmemorypool", |_rpc_params, rpc_context| async move {
+        rpc_context.get_memory_pool().map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    module.register_async_method("gettransaction", |rpc_params, rpc_context| async move {
+        let id = rpc_params.parse::<[N::TransactionID; 1]>()?[0];
+        rpc_context.get_transaction(id).map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    module.register_async_method("gettransition", |rpc_params, rpc_context| async move {
+        let id = rpc_params.parse::<[N::TransitionID; 1]>()?[0];
+        rpc_context.get_transition(id).map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    module.register_async_method("getconnectedpeers", |_rpc_params, rpc_context| async move {
+        rpc_context.get_connected_peers().map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    module.register_async_method("getnodestate", |_rpc_params, rpc_context| async move {
+        rpc_context.get_node_state().map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    module.register_async_method("sendtransaction", |rpc_params, rpc_context| async move {
+        let string = std::mem::take(&mut rpc_params.parse::<[String; 1]>()?[0]);
+        rpc_context.send_transaction(string).map_err(JsonrpseeError::to_call_error).await
+    })?;
+
+    // Private methods.
+
+    // "createtransaction" => {
```

Comment:
Kept for reference?

Reply:
they were already commented out - presumably temporarily - so I preserved them so it's later easier to bring them back